MovieApp/PresentationLayer/Presenters/HomeScreenPresenter.swift

```diff
@@ -1,11 +1,35 @@
 class HomeScreenPresenter {
 
-    private let movies = MockDataService.movies
+    private var movies: [MovieViewModel] = []
+    private let movieClient: MovieClient
+
+    var delegate: HomeScreenViewController!
+
+    init(baseUrl: String) {
+        movieClient = MovieClient(baseApiClient: BaseApiClient(baseUrl: baseUrl))
+    }
 
     public var numberOfMovies: Int {
         movies.count
     }
 
+    func fetchMovies() {
+        movieClient.fetchPopularMovies(completionHandler: { [weak self]
+            (result: Result<PopularMoviesModel, RequestError>) -> Void in
```

Comment:
Do you really need to specify the type? Can you put just `{ [weak self] result in`?